app/objects/player.py

```diff
@@ -522,6 +532,20 @@ async def unrestrict(self, admin: "Player", reason: str) -> None:
             {"from": admin.id, "to": self.id, "action": "unrestrict", "msg": reason},
         )
 
+        if not self.stats:
+            async with app.state.services.database.connection() as db_conn:
+                await self.stats_from_sql_full(db_conn)
+
+        for idx, mode_stat in enumerate(self.stats.values()):
```

Comment:
stats keys are modes i believe, could we do `for mode, stats in self.stats.items():`? (don't have time to check atm, at work)